biocypher_metta/metta_writer.py

```diff
@@ -96,15 +97,22 @@ def write_nodes(self, nodes, path_prefix=None, create_dir=True):
                     pathlib.Path(f"{self.output_path}/{path_prefix}").mkdir(parents=True, exist_ok=True)
         else:
             file_path = f"{self.output_path}/nodes.metta"
+        node_freq = Counter()
```

Comment:
Why not create a parent writer class and move the counting or at least the field there? I see the same code for counting across all writes

Reply:
Done! I created a parent writer class to centralize the counting logic. Thanks for the suggestion!